analyzers/tests/SonarAnalyzer.UnitTest/Helpers/SonarAnalysisContextTest.ShouldAnalyzeGenerated.cs

```diff
@@ -19,192 +19,139 @@
 
-        // We can't use Mock<SourceText> because SourceText is an abstract class
-        private class DummySourceText : SourceText
-        {
-            private readonly string textToReturn;
-            public int ToStringCallCount { get; private set; }
+        sonarLintXml.ToStringCallCount.Should().Be(2, "file should be read once per language");
 
-            public DummySourceText(string textToReturn)
-            {
-                this.textToReturn = textToReturn;
-            }
+        // Read again to check caching
+        sut.ShouldAnalyzeGenerated(compilationVB, options).Should().BeTrue();
 
-            public override string ToString()
-            {
-                ToStringCallCount++;
-                return textToReturn;
-            }
+        sonarLintXml.ToStringCallCount.Should().Be(2, "file should not have been read again");
+    }
 
-            #region Abstract methods
+    private static SonarAnalysisContext CreateSut() =>
+        new(Mock.Of<AnalysisContext>(), Enumerable.Empty<DiagnosticDescriptor>());
+
+    private static DummySourceText CreateSonarLintXml(bool analyzeGeneratedCSharp) =>
+        new($"""
+            <?xml version="1.0" encoding="UTF-8"?>
+            <AnalysisInput>
+                <Settings>
+                    <Setting>
+                        <Key>dummy</Key>
+                        <Value>false</Value>
+                    </Setting>
+                    <Setting>
+                        <Key>sonar.cs.analyzeGeneratedCode</Key>
+                        <Value>{analyzeGeneratedCSharp.ToString().ToLower()}</Value>
+                    </Setting>
+                    <Setting>
+                        <Key>sonar.vbnet.analyzeGeneratedCode</Key>
+                        <Value>true</Value>
+                    </Setting>
+                </Settings>
+            </AnalysisInput>
+            """);
+
+    private static AnalyzerOptions CreateOptions(SourceText sourceText, string path = @"ResourceTests\SonarLint.xml") =>
+        new(ImmutableArray.Create(MockAdditionalText(sourceText, path).Object));
+
+    private static Mock<AdditionalText> MockAdditionalText(SourceText sourceText, string path = @"ResourceTests\SonarLint.xml")
+    {
+        var additionalText = new Mock<AdditionalText>();
+        additionalText.Setup(x => x.Path).Returns(path);
+        additionalText.Setup(x => x.GetText(System.Threading.CancellationToken.None)).Returns(sourceText);
+        return additionalText;
+    }
 
-            public override char this[int position] => throw new System.NotImplementedException();
+    private static Compilation GetDummyCompilation(AnalyzerLanguage language) =>
+        TestHelper.Compile(string.Empty, false, language).Model.Compilation;
 
-            public override Encoding Encoding => throw new System.NotImplementedException();
+    // We can't use Mock<SourceText> because SourceText is an abstract class
```

Comment:
There is something fishy with this comment, `AdditionalText` is also an abstract class and it seems to me that it is "Mocked" without any problem.

Reply:
I throw it away. As having a dummy class is more handy (reusable) anyway

Reply:
I prefer mocking, and that is only a preference. As the class was already there, I am fine with only removing the incorrect comment.